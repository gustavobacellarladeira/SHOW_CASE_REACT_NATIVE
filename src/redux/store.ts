import { applyMiddleware, combineReducers, createStore } from "redux";
import thunk from "redux-thunk";
import { themeReducer } from "./themeReducer";

const store = createStore(
  combineReducers({ themeReducer }),
  applyMiddleware(thunk)
);

export { store };
