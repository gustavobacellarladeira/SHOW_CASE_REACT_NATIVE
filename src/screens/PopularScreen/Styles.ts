import styled from "styled-components/native";

export const Container = styled.View`
  flex: 1;
  width: 100%;
`;

export const ContainerSection = styled.View`
  flex: 1;
  align-items: center;
  justify-content: space-around;
`;
export const SectionPrimary = styled.View`
  height: 10%;
  background-color: ${(props) => props.theme.colors.primary};
  align-items: center;

  justify-content: center;
`;
export const TitleSectionPrimary = styled.Text`
  color: ${(props) => props.theme.PRIMARY_TEXT_COLOR};
  font-size: ${(props) => props.theme.FONT_SIZE_TITLE};

  text-align: left;
`;
export const SectionSecondary = styled.View`
  background-color: ${(props) => props.theme.colors.secondary};
  flex: 1;
`;
export const SkeletonContainer = styled.View`
  flex: 1;
  margin-horizontal: 10px;
  min-width: 140px;
  min-height: 160px;
  width: 100%;
  height: 100%;
  align-items: center;
  justify-content: center;
  overflow: hidden;
`;
export const FlatlistContainer = styled.View`
  padding: 10px;
  width: 100%;
`;

export const FlatListCard = styled.View`
  background-color: ${(props) => props.theme.colors.primary};
  border-radius: 10px;
  flex-direction: row;
  justify-content: space-between;
  padding: 10px;
`;

export const TitleFlatlistContainer = styled.View`
  width: 60%;
  align-items: center;
  background-color: ${(props) => props.theme.colors.primary};
  padding-top: 10px;
  border-radius: 10px;
`;
export const TitleFlatlist = styled.Text`
  color: ${(props) => props.theme.PRIMARY_TEXT_COLOR};
  font-size: ${(props) => props.theme.FONT_SIZE_TITLE};

  text-align: left;
`;

export const ImageContainer = styled.View`
  background-color: ${(props) => props.theme.colors.primary};
  border-radius: 10px;
  width: 40%;
`;
export const EndList = styled.View`
  align-items: center;
  justify-content: center;
  background-color: ${(props) => props.theme.colors.tertiary};
  padding: 30px;
`;
export const EndListText = styled.Text`
  color: ${(props) => props.theme.colors.primary};
  font-size: ${(props) => props.theme.fonts.size};
  text-align: center;
`;

export const EmptyList = styled.View`
  min-height: 100%;
  align-items: center;
  justify-content: center;
  background-color: ${(props) => props.theme.colors.secondary};
  padding: 30px;
`;

export const EmptyListText = styled.Text`
  color: ${(props) => props.theme.PRIMARY_TEXT_COLOR};

  font-size: ${(props) => props.theme.fonts.size};
  text-align: center;
`;
