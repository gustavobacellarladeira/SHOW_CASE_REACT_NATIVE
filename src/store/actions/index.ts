export { REMOVE_MOVIE, SET_MOVIE } from "./movies";
