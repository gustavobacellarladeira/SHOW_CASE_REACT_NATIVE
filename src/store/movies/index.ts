export { moviePopularReducer } from "./popular/reducer";
export { movieTrendingReducer } from "./trendings/reducer";
