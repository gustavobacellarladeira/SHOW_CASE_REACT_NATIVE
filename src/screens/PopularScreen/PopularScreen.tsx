import React, { useCallback, useEffect, useState } from "react";
import {
  ActivityIndicator,
  Button,
  FlatList,
  Text,
  TouchableOpacity,
  View,
} from "react-native";

import {
  Container,
  ContainerSection,
  FlatlistContainer,
  SectionPrimary,
  SectionSecondary,
  SkeletonContainer,
  TitleFlatlistContainer,
  TitleSectionPrimary,
} from "./Styles";

import { Skeleton } from "../../components";
import { useMovies } from "../../store/movies/hook";
import { FAKE_MOVIES, LIMIT_MOVIES } from "../../constants/movies";

export const PopularScreen = () => {
  const { popular, loadingMovies, getPopularMovies } = useMovies();
  const [page, setPage] = useState<number>(1);
  const [isEndList, setIsEndList] = useState<boolean>(false);

  useEffect(() => {
    console.log("Movies -->", popular.length);
    console.log("loadingMoviesMovies -->", loadingMovies);
  }, [loadingMovies]);

  useEffect(() => {
    console.log("Movies -->", popular.length);
  }, [popular.length]);

  const handlerLoadMorePopularMovies = useCallback(() => {
    let moviesLength = popular.length;
    // falta uma variavel na api para saber se tem mais para loading ou nao...
    // por isso irei colocar um maximo no length de movies
    if (moviesLength >= LIMIT_MOVIES) {
      console.log("Não há mais filmes para carregar");
      setIsEndList(true);
      return;
    } else {
      getPopularMovies(page + 1);
      setPage((prev) => prev + 1);
    }
  }, [page, getPopularMovies, popular.length]);

  return (
    <Container>
      {/* CABEÇALHO DO CONTEÚDO */}
      <SectionPrimary>
        <TitleSectionPrimary>Lista de filmes </TitleSectionPrimary>
        <Button title="Popular" onPress={() => {}} />
        <Button title="Trending" onPress={() => {}} />
      </SectionPrimary>
      {/* LISTAS DE FILMES */}
      <SectionSecondary>
        <FlatList
          style={{}}
          contentContainerStyle={{}}
          data={loadingMovies ? FAKE_MOVIES : popular}
          renderItem={({ item }) => (
            <FlatlistContainer>
              {loadingMovies ? (
                <Skeleton borderRadius={5} />
              ) : (
                <TitleSectionPrimary> {item.title} </TitleSectionPrimary>
              )}
            </FlatlistContainer>
          )}
          keyExtractor={(item) =>
            loadingMovies ? item.toString() : item.ids.slug + item.ids.trakt
          }
          onEndReached={handlerLoadMorePopularMovies}
          onEndReachedThreshold={0.5}
          ListFooterComponent={
            loadingMovies ? (
              <View></View>
            ) : (
              <>{isEndList && <Text>Não há mais filmes para carregar</Text>}</>
            )
          }
          showsVerticalScrollIndicator={false}
          showsHorizontalScrollIndicator={false}
          ListEmptyComponent={
            <Text>
              {loadingMovies ? "Carregando..." : "Nenhum filme encontrado"}
            </Text>
          }
        />
      </SectionSecondary>
    </Container>
  );
};

// <TitleSectionPrimary> {item.title} </TitleSectionPrimary>
