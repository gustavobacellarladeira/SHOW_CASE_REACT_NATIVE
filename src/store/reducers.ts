import { combineReducers } from "redux";
import { themeReducer } from "./theme/reducer";
import { movieReducer } from "./movies/index";

export const reducer = combineReducers({
  movies: movieReducer,
  theme: themeReducer,
});
