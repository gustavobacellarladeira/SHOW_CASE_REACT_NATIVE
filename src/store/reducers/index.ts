import { combineReducers } from "redux";
import { movieReducer } from "./movies";

export const reducer = combineReducers({
  movies: movieReducer,
});
