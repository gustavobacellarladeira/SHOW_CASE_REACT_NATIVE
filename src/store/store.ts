import { applyMiddleware, createStore } from "redux";
import { reducer } from "./reducers";
import promiseMiddleware from "redux-promise-middleware";
import promise from "redux-promise-middleware";
import thunk from "redux-thunk";

applyMiddleware(promiseMiddleware)(createStore);

const store = createStore(reducer, {}, applyMiddleware(promise));
export { store };
