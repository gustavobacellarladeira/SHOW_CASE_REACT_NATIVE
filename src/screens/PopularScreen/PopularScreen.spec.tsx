import { store } from "../../store/store";
import React from "react";
import { Provider } from "react-redux";
import { render, fireEvent, act } from "@testing-library/react-native";
import { ThemeProvider, DefaultTheme } from "styled-components";

import { PopularScreen } from "./PopularScreen";

const MOCK_THEME: DefaultTheme = {
  mode: "dark",
  PRIMARY_BACKGROUND_COLOR: "#333333",
  PRIMARY_TEXT_COLOR: "#ffffff",
  FONT_SIZE_TITLE: "20px",
  fonts: {
    color: "#ffffff",
    size: "20px",
  },
  colors: {
    primary: "#222831",
    secondary: "#393E46",
    tertiary: "#D65A31",
  },

  tabBottomBar: {
    backgroundColor: "#333",
    focused: "#D65A31",
    inactive: "#ffffff",
  },
};

const rendererWithRedux = (component: any) => {
  return render(
    <Provider store={store}>
      <ThemeProvider theme={MOCK_THEME}>{component}</ThemeProvider>
    </Provider>
  );
};

describe("PopularScreen testing...", () => {
  it("renders correctly", () => {
    const tree = rendererWithRedux(<PopularScreen />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
