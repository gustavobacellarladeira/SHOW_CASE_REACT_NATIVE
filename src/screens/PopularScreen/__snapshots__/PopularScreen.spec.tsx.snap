// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularScreen testing... renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#000",
          "height": "10%",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 20,
            "textAlign": "left",
          },
        ]
      }
    >
      Popular Movies
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#000",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={[Function]}
      ListFooterComponent={false}
      contentContainerStyle={Object {}}
      data={
        Array [
          1,
          2,
          3,
          4,
          5,
          6,
          7,
          8,
          9,
          10,
          11,
          12,
          13,
          14,
          15,
          16,
          17,
          18,
          19,
          20,
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0.5}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      style={Object {}}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 5,
                  "flex": 1,
                  "opacity": 0.2,
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 5,
                  "flex": 1,
                  "opacity": 0.2,
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 5,
                  "flex": 1,
                  "opacity": 0.2,
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 5,
                  "flex": 1,
                  "opacity": 0.2,
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 5,
                  "flex": 1,
                  "opacity": 0.2,
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 5,
                  "flex": 1,
                  "opacity": 0.2,
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 5,
                  "flex": 1,
                  "opacity": 0.2,
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 5,
                  "flex": 1,
                  "opacity": 0.2,
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 5,
                  "flex": 1,
                  "opacity": 0.2,
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 5,
                  "flex": 1,
                  "opacity": 0.2,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "height": 0,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;
